pkg/daemon/daemon.go

```diff
@@ -38,6 +38,18 @@ func New(cfg *config.EnvConfig) (srv *Daemon, err error) {
 
 	r := mux.NewRouter()
 
+	// Authentication
+	r.Use(func(next http.Handler) http.Handler {
```

Comment:
We should add this handler only if there are tokens in the cfg.

Reply:
Right, I added check for len, but inside the handler. You're right that it should be for the handler itself.